### DISCUSSION TITLE
FLASK==3.0.3/ Werkzeug==3.0.0 Cannot import name 'url_quote' from 'werkzeug.urls'

### DISCUSSION THREAD

**Discussion by esefx** [pallets/flask#5506] [Q&A]
I keep getting the error:  Cannot import name 'url_quote' from 'werkzeug.urls'. I'm using python 3.12

I've tried the following combinations with no success: 
Flask==3.0.3 
Werkzeug==3.0.0

Flask==2.2.2 
Werkzeug==2.2.2 

Flask == 2.0.1 
Werkzeug==2.2.2

flask>=2.0  
Werkzeug>=2.0

flask==2.01
Werkzeug==2.2.2

Please Help!

**Comment by davidism** [comment]
Be sure to include a minimal reproducible example, including the full traceback when you get an error. Using the latest versions of Flask and Werkzeug do not have this problem. It's likely this error is coming from an extension, either an old version or one that has not been updated to work with modern versions of Flask.

**Reply by esefx** [reply]
Ah, thank you for this comment. I was able to debug with it and realized that the problem was that when I was running FLASK_ENV=development flask run it was accessing a flask installed in my home directory not the project folder. I was able to fix my errors by updating the flask on my computers root directory.
